app/models/recipe.rb

```diff
@@ -46,4 +48,14 @@ def total_grain_weight
     grains.sum(:weight)
   end
 
+  def reject_blank_grains_and_hops
+    [grains, hops].each do |grain_or_hop|
+      grain_or_hop.each do |ingredient|
+        if ingredient.weight.blank? && ingredient.name.blank?
+          ingredient.mark_for_destruction
+        end
+      end
+    end
+  end
```

Comment:
Nice work on `mark_for_destruction` (I'd not encountered that, thanks for the tip).

It's starting to look like the `Grain`, `Hop`, and `Yeast` models might benefit from having their own `blank?` methods that wrap the `weight.blank? && name.blank?` etc. checks.